### PR TITLE
revert gradle-it removal (#2214), with fix

### PR DESCRIPTION
**Pull request by vorburger**

### CODE REVIEW

**Comment by vorburger** [comment]
While this PR as-is currently makes `devtools/gradle-it` build again, the Gradle built JAR fails to run:

```
$ java -jar build/gradle-it-runner.jar 
java.lang.NoClassDefFoundError: org/wildfly/common/format/Printf
	at io.quarkus.runtime.logging.LoggingSetupTemplate.initializeLogging(LoggingSetupTemplate.java:59)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit4.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit4.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:100)
	at io.quarkus.runtime.Application.run(Application.java:212)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.ClassNotFoundException: org.wildfly.common.format.Printf
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 7 more
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:100)
	at io.quarkus.runtime.Application.run(Application.java:212)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.NoClassDefFoundError: org/wildfly/common/format/Printf
	at io.quarkus.runtime.logging.LoggingSetupTemplate.initializeLogging(LoggingSetupTemplate.java:59)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit4.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit4.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
	... 3 more
Caused by: java.lang.ClassNotFoundException: org.wildfly.common.format.Printf
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 7 more
```

does anyone have a clue what may be still missing?

This used to work, but seems to have broken in recent weeks.

**Comment by stalep** [comment]
Hi Michael, glad to have you back! :)
I just noticed the issue, give me some time to look at it.

**Comment by FroMage** [comment]
@stalep ping? :)

**Comment by emmanuelbernard** [comment]
Ping again @stalep

**Review by stalep — COMMENTED** [review]
Really sorry for the long time time to review it!
It works and looks fine.
I'm just worried about it breaking the release process, but from what I can see there is no hard-coding of versions @gsmet or @cescoffier?

**Review by stalep — APPROVED** [review]

File: devtools/gradle-it/pom.xml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-deployment</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <version>${project.version}</version>
```
Comment (gsmet) [inline devtools/gradle-it/pom.xml]: Aren't all these versions defined by the parent?

File: devtools/pom.xml [hunk 2]
```diff
@@ -41,6 +41,7 @@
         <module>common</module>
         <module>maven</module>
         <module>gradle</module>
+        <module>gradle-it</module>
```
Comment (gsmet) [inline devtools/pom.xml]: I would prefer having it with the other integration tests.

**Review by gsmet — COMMENTED** [review]
I added a couple of comments.

**Comment by geoand** [comment]
I just tried this PR against master and it did catch what #3173 fixes

**Comment by geoand** [comment]
PR tried with the patched `master` and the tests pass as expected. @vorburger can you please address the comment of @gsmet to move the integration tests so we can then merge?

**Comment by dmlloyd** [comment]
Any update?

**Comment by geoand** [comment]
I'm going to close this since #4579 supersedes it

**Pull request closed without merging**